### PR TITLE
[KernelGen][Ascend] Add bernoulli operator with Triton kernel

### PR DESCRIPTION
**Pull request by LoserCheems**
# [KernelGen][Ascend] Add bernoulli operator with Triton kernel

## Summary
Add `bernoulli` operator Triton kernel implementation for the Ascend backend. Draws binary random numbers (0 or 1) from a Bernoulli distribution with given probabilities. Supports both tensor-probability (`bernoulli`) and scalar-probability (`bernoulli.p`) variants.

## Testing
- 24 parametrized tests passed (shapes × dtypes)
- Validated output contains only 0/1 values
- Tested with float16, float32, bfloat16
- Reference: PyTorch CPU aten dispatch

## Performance

**bernoulli (tensor probability):**

| dtype | shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|---|
| float16 | [4096, 4096] | 168.4 | 32.8 | 5.14x |
| float16 | [1024, 1024, 1024] | 11311.8 | 1989.4 | 5.69x |
| float32 | [4096, 4096] | 86.1 | 33.0 | 2.61x |
| float32 | [1024, 1024, 1024] | 5835.7 | 2005.4 | 2.91x |
| bfloat16 | [4096, 4096] | 231.4 | 32.8 | 7.06x |
| bfloat16 | [1024, 1024, 1024] | 15591.0 | 1991.5 | 7.83x |

**bernoulli.p (scalar probability):** Performance on par with torch (1.0x speedup).

Note: Small tensors ([64, 64]) are slower due to kernel launch overhead.

## Files Changed
- `src/flag_gems/runtime/backend/_ascend/ops/bernoulli.py`: Triton kernel implementation
- `tests/test_bernoulli.py`: Accuracy tests (appended to existing bernoulli_ tests)
- `benchmark/test_bernoulli.py`: Performance benchmarks (appended to existing bernoulli_ benchmarks)
- `src/flag_gems/runtime/backend/_ascend/ops/__init__.py`: Register import and `__all__`
- `conf/operators.yaml`: Add operator entry (kind: Tensor, stage: alpha 5.1)

### CODE REVIEW

**Review by Dongxu-H — CHANGES_REQUESTED** [review]
## Code Review: Bernoulli Distribution Implementation

### Overview
This PR adds `bernoulli` and `bernoulli_p` operators for sampling from Bernoulli distribution. Supports both random sampling from tensor values and direct probability sampling.

### 🔴 Critical Issues

1. **Random Seed/State Management**
   - File: `src/flag_gems/ops/bernoulli.py`
   - No proper RNG state initialization for Triton kernels
   - Multiple calls with same tensor will produce identical results
   - **Recommendation**: Implement proper seeding using `torch.rand` or Triton's random utilities

2. **Numerical Precision**
   - Using `tl.rand()` which returns float32, but comparison with `probs` may have precision issues
   - Edge case: when `prob` is exactly 0.0 or 1.0, should always return 0 or 1 respectively
   - **Recommendation**: Add explicit checks for prob in {0, 1}

3. **Non-Deterministic Behavior Across Backends**
   - Different GPU architectures may have different random number implementations
   - Results may not be reproducible across devices
   - **Recommendation**: Document this limitation or use deterministic seeding

### 🟡 Suggestions

4. **Missing Input Validation**
   - No check for `probs` outside [0, 1] range
   - Should raise ValueError for invalid probabilities
   - **Recommendation**: Add `torch.clamp(probs, 0, 1)` or validation

5. **Performance Optimization**
   - The `bernoulli_p` kernel could be optimized by avoiding the `tl.sqrt` in random number generation
   - Consider pre-computing random values in shared memory for better cache utilization

6. **Test Coverage**
   - Missing tests for:
     - Empty tensors
     - Very large tensors (statistical distribution test)
     - Edge cases: prob = 0.0, 1.0
     - Multi-dimensional tensors
     - Non-contiguous tensors

7. **Statistical Correctness**
   - Should test that samples follow correct Bernoulli distribution
   - **Recommendation**: Add chi-squared test or similar statistical validation

8. **Documentation**
   - Missing docstrings for `bernoulli` and `bernoulli_p`
   - Should document the difference between the two functions

### ✅ Strengths

- Clean kernel implementation
- Proper use of `@libentry()` decorator
- Good integration with existing codebase structure

### Questions

- What's the expected behavior when `probs` tensor has values outside [0, 1]?
- Is there a performance difference between `bernoulli` (tensor-based) and `bernoulli_p` (scalar probability)?
- Should this support `generator` parameter for reproducible random sampling (like PyTorch)?
- Are there plans to support `out` parameter for in-place sampling?

### Additional Recommendations

1. **Statistical Test**
   Add a test to verify distribution:
   ```python
   def test_bernoulli_distribution():
       p = 0.3
       samples = bernoulli_p(torch.empty(10000), p)
       assert abs(samples.mean().item() - p) < 0.02  # Statistical check
   ```

2. **Deterministic Mode**
   Consider supporting `torch.use_deterministic_algorithms()` for reproducibility in testing/debugging.
